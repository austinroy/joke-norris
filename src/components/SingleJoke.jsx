import React, { Component } from 'react';
import axios from 'axios'

class SingleJoke extends Component {
	state = {
		joke: {}
	}

  componentDidMount = () => {
		const category = this.props.match.params.category;

    axios.get(`https://api.chucknorris.io/jokes/random?category=${category}`)
    .then(res =>{
        if(res.status === 200) {
            const joke = res.data;
            this.setState({joke : joke})
        }
      }
    )
	}
	
	renderJoke = joke => {
    const category = this.props.match.params.category
    if (joke){
      return(
        <div key={joke.id}>
          <div>
            <h1>{category}</h1>
            <p>{joke.value}</p>
          </div>
        </div>
      )
    } else {
      return (
        <div>
          Joke Not Loaded
        </div>
      )
		}
	}

	render(){
		const { joke } = this.state;
		return(
			<div className="singlejoke">
				{this.renderJoke(joke)}
			</div>
		)
	}
}

export default SingleJoke;