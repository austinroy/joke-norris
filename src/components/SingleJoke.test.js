import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { SingleJoke } from './SingleJoke';

afterEach(cleanup);

const props = {
  fetchJoke: jest.fn(),
  loading: false,
  joke: 'Chuck Norris writes code that optimizes itself.',
  match: {
    params: {
      category: 'dev'
    }
  }
};

const propsFailed = {
  fetchJoke: jest.fn(),
  loading: true,
  match: {
    params: {
      category: 'dev'
    }
  }
};

test('renders the expected joke', () => {
  const container = render(<SingleJoke {...props} />);
  expect(container).toMatchSnapshot();
});

test('renders joke failure ui joke is not loaded', () => {
  const container = render(<SingleJoke {...propsFailed} />);
  expect(container).toMatchSnapshot();
});
