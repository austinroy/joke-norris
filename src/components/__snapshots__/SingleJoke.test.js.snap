// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders joke failure ui joke is not loaded 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ui active inline loader"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ui active inline loader"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the expected joke 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ui raised segment"
        >
          <a
            class="ui red massive ribbon label"
            style="text-transform: uppercase;"
          >
            dev
          </a>
          <div>
            <div>
              <p
                class="joke"
              />
            </div>
          </div>
          <a
            class="ui icon primary right labeled button"
            href="/"
            role="button"
          >
            <i
              aria-hidden="true"
              class="home icon"
            />
            Choose Category
          </a>
          <button
            class="ui teal icon positive right labeled button"
          >
            Next
            <i
              aria-hidden="true"
              class="right arrow icon"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ui raised segment"
      >
        <a
          class="ui red massive ribbon label"
          style="text-transform: uppercase;"
        >
          dev
        </a>
        <div>
          <div>
            <p
              class="joke"
            />
          </div>
        </div>
        <a
          class="ui icon primary right labeled button"
          href="/"
          role="button"
        >
          <i
            aria-hidden="true"
            class="home icon"
          />
          Choose Category
        </a>
        <button
          class="ui teal icon positive right labeled button"
        >
          Next
          <i
            aria-hidden="true"
            class="right arrow icon"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
